src/main/java/net/dv8tion/jda/core/entities/Message.java

```diff
@@ -789,19 +787,9 @@ public boolean download(File file)
             InputStream in = null;
             try
             {
-                URL url = new URL(getUrl());
-                URLConnection con;
-                if (jda.getGlobalProxy() == null)
-                {
-                    con = url.openConnection();
-                }
-                else
-                {
-                    con = url.openConnection(new Proxy(Proxy.Type.HTTP,
-                            new InetSocketAddress(jda.getGlobalProxy().getAddress(), jda.getGlobalProxy().getPort())));
-                }
-                con.addRequestProperty("user-agent", Requester.USER_AGENT);
-                in = con.getInputStream();
+                Request request = new Request.Builder().addHeader("user-agent", Requester.USER_AGENT).url(getUrl()).build();
+                Response response = jda.getRequester().getHttpClient().newCall(request).execute();
+                in = response.body().byteStream();
```

Comment:
https://github.com/DV8FromTheWorld/JDA/blob/experimental/message-rw/src/main/java/net/dv8tion/jda/core/entities/Message.java#L977-L1006 this can easily be made public in `experimental/message-rw` later on.